src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -362,6 +372,50 @@ def validate_args(args: Tuple[str, ...]):
         raise InvalidSetupPyArgs("Cannot use the `upload` or `register` setup.py commands")
 
 
+@rule
+async def package_python_dist(
+    field_set: PythonDistributionFieldSet,
+    python_setup: PythonSetup,
+) -> BuiltPackage:
+
+    transitive_targets = await Get(TransitiveTargets, Addresses([field_set.address]))
+    exported_target = ExportedTarget(transitive_targets.roots[0])
+    interpreter_constraints = PexInterpreterConstraints.create_from_compatibility_fields(
+        (
+            tgt[PythonInterpreterCompatibility]
+            for tgt in transitive_targets.dependencies
+            if tgt.has_field(PythonInterpreterCompatibility)
+        ),
+        python_setup,
+    )
+    chroot = await Get(
+        SetupPyChroot,
+        SetupPyChrootRequest(exported_target, py2=interpreter_constraints.includes_python2()),
+    )
+
+    # If commands were provided, run setup.py with them; Otherwise just dump chroots.
+    if exported_target.target.has_field(SetupPyCommandsField):
```

Comment:
That this solely checks "Is the field registered?" This means that every single `python_distribution` will use it. Do you mean to instead check that it has the field and that the field has a value set?

As it stands, I don't think the `else` block can ever trigger.

Reply:
Ah yes, good catch